### ISSUE TITLE
Allow disabling remote build from the command line when using nix-daemon

### ISSUE THREAD

**Issue by k0001** [NixOS/nix#1118]
When using `nix-daemon` to build stuff remotely (for example, when setting `nix.distributedBuilds = true` in `/etc/nixos/configuration.nix`), it doesn't seem to be possible to temporarily force the builds to happen locally when calling `nix-build`. 

What I would like is to be able to say something like `nix-build --build-locally foo` and have `foo` be built locally, overriding whatever default behavior was set for `nix-daemon`. This is useful because sometimes one wants to avoid having to upload a lot of dependencies to the remote end just to build a small package. 

If there already is a way to temporarily disable this other than setting `nix.distributedBuilds = false` and calling `nixos-rebuild switch`, please let me know :)

**Comment by vcunat** [comment]
I use `sudo mv /etc/nix/machines{,.}`, but that's not really a nice way. Then I sometimes copy the contents of the file and edit it in-place.

**Comment by zimbatm** [comment]
I just hit the same problem and found an undocumented option as `--no-build-hook` that does what you want.

**Comment by k0001** [comment]
Thanks!

**Comment by volth** [comment]
The ticket must be reopened, as the proposed solution (```--no-build-hook``` option) has been removed from modern Nix and currently there is no solution (besides ```sudo mv /etc/nix/machines```)

**Comment by zimbatm** [comment]
Agreed. Nix 2.0 accepts `--builders` to pass additional builders on the command-line. It would be nice if is also supported a `--no-builders` argument or `--builders ""`

**Comment by LnL7** [comment]
`--builders ''` works and is the reason why --no-build-hook was removed AFAIK.

**Comment by shlevy** [comment]
@LnL7 Doesn't work as an untrusted user though.

**Comment by shlevy** [comment]
Would be nice if `builders` could be a subset of `trusted-builders` like `substituters`

**Comment by volth** [comment]
> ```--builders ''``` works and is the reason why --no-build-hook was removed AFAIK.

Not for me.
```--builders ''``` builds on remote server as without ```--builders```

**Comment by zimbatm** [comment]
it would be nice if `--builders ''` failed instead of being essentially a no-op if the user is not trusted

**Comment by JohnAZoidberg** [comment]
It would also nice to be able to force `nixos-rebuild` to build locally. I currently have the problem that I have specified `nix.distributedBuilds = true` in my config, but to set it to false and rebuild takes way too long because I am on a slow connection. I only got around it by forcing the connection to the build server to fail.

**Comment by LnL7** [comment]
`--option builders ''` should also work for nixos-rebuild.

**Comment by timokau** [comment]
Can somebody re-open this? If I understand correctly, there is currently no solution as a non-trusted user.

**Comment by Mic92** [comment]
@LnL7 I implemented it here: https://github.com/NixOS/nixpkgs/pull/53801

**Comment by dtzWill** [comment]
FWIW empty builders only worked as of #2448, so depending on version you're using might not be the behavior observed.  `nixos-rebuild` already accepts builders option IIRC, but it's `--option builders ""`, wouldn't mind seeing that become `--builders ""`.

**Comment by DavHau** [comment]
I'm using nix 2.3.6 and none of the proposed flags work with the `nix-build` command. Can we re-open this?

**Comment by Ericson2314** [comment]
Yes, let's reopen this

> Would be nice if `builders` could be a subset of `trusted-builders` like `substituters`

@shlevy is exactly right.

**Comment by DavHau** [comment]
If anybody is looking for a workaround to use with `nix-build`. I figured setting `--builders` to something invalid like `--builders "42"` leads to a local-only build. But I don't know if this works in all cases.

**Comment by mohe2015** [comment]
I also first thought this was happening but after removing `/etc/nix/machines` (or moving it) **and** using 
`--option substitute false` I could finally build (I followed the tutorial in the Wiki and the error message doesn't mention the problem is the substitute and not the builder).

**Comment by grwlf** [comment]
> after removing `/etc/nix/machines` (or moving it)

For me this was enough to temporary stop distributing

**Comment by Ericson2314** [comment]
We should have a test for this. now that we have `--foo` vs `--extra-<foo>`, `--builders ''` I would think should work.

**Comment by MakiseKurisu** [comment]
I was figuring out how to set up remote builders and misconfigured my machine, which blocked my rebuild. `--builders ""` does not work, but `--builders "@/dev/null"` worked.

**Comment by thislooksfun** [comment]
Using `--builders ''` worked for me just now.
